### ISSUE TITLE
Chocolatey bug: Nu crash 4294967295 with Ctrl+C when stopping Node process under VS Code's terminal (Windows)

### ISSUE THREAD

**Issue by Stefouch** [nushell/nushell#6428]
### Describe the bug

When typing CTRL+C in my Visual Studio Code's integrated terminal to interrupt a NodeJS process, Nu crashes with the following message:
```sh
The terminal process "C:\ProgramData\chocolatey\bin\nu.exe" terminated with exit code: 4294967295.
```

Exit code: `4294967295`

If I run the Nu terminal directly from the folder, it works fine.

### How to reproduce

1. Open VS Code
2. Open the integrated terminal with Nu
3. `> node .` to launch a NodeJS process (in my situation, it launches a Discord bot)
4. Type CTRL+C to stop the process
5. The integrated terminal closes (it shouldn't) and pops-up an error message.

### Expected behavior

The integrated terminal should stay open, with Nu ready for the next command input.

### VS Code integrated terminal:
![image](https://user-images.githubusercontent.com/47865544/187027417-0dee9cd5-d390-4501-ae01-2e742f49ded4.png)

### Crash pop-up:
![image](https://user-images.githubusercontent.com/47865544/187026611-1875ad4e-4cc2-4dc2-80c2-05e7fa028d64.png)

### Expected behavior:
![image](https://user-images.githubusercontent.com/47865544/187027374-5ed50c7e-9fb9-4610-b74c-9eb8f6b3d56d.png)

### Configuration

| key | value |
| --- | --- |
| version            | 0.67.0                                          |
| branch             | feat/discord-v14                                |
| commit_hash        | 91976988f032276ca48faff2d1ac13c079eda836        |
| build_os           | windows-x86_64                                  |
| build_target       | x86_64-pc-windows-msvc                          |
| rust_version       | rustc 1.63.0 (4b91a6ea7 2022-08-08)             |
| rust_channel       | stable-x86_64-pc-windows-msvc                   |
| cargo_version      | cargo 1.63.0 (fd9c4297c 2022-07-01)             |
| pkg_version        | 0.67.0                                          |
| build_time         | 2022-08-17 15:20:56 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                 

### Additional context

I understand that finding a solution to this issue looks like a nightmare with how many variables there are. Also, it might be VS Code's fault and not Nu.

Here are some other relevant information:
| pkg | version | comment |
| --- | --- | --- |
| Windows | 10  | 21H2 |
| Chocolatey | 1.1.0 | Used to install Nu & Starship |
| Starship | 1.9.1 | Custom prompt for Nu |
| VS Code | 1.70.2 |
| Node | 16.16.0 | LTS
| discord.js | 14.3.0 | JS popular module for communication with Discord's API V10 |

**Comment by rgwood** [comment]
Thanks for the bug report! Unfortunately I can't reproduce this:
![image](https://user-images.githubusercontent.com/26268125/187048400-c87bfc37-aa5c-4ec4-a638-b53564974d62.png)

Some wild guesses as to what the cause might be:

- Something to do with [your Chocolatey shim/stub](https://docs.chocolatey.org/en-us/features/shim) (I built my copy of Nu locally)
- Node version mismatch
- Windows 10 behaving differently than 11 (unlikely, but there were a number of terminal-related improvements for 11)

## Version Info

Windows 11 (v10.0.22000.856)
Node v18.8.0
VS Code 1.70.2

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.67.1                                   |
| branch             |                                          |
| commit_hash        | 918ec9daa839d55f1e607f588313e55a76b60a37 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.61.0 (fe5b13d68 2022-05-18)      |
| rust_channel       | 1.61.0-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.61.0 (a028ae42f 2022-04-29)      |
| pkg_version        | 0.67.1                                   |
| build_time         | 2022-08-25 12:18:00 -07:00               |
| build_rust_channel | release                                  |
| features           | database, default, trash, which, zip     |
| installed_plugins  |                                          |

**Comment by Stefouch** [comment]
OK thank you! Your message actually helped me to solve this problem.

I first tested your simple code and it was crashing too. So it was definitely not related to my Discord bot's code.

Then I read the documentation you shared about Chocolatey's shim and was *"Hu? What is that?"*
From there, I investigated and found that, indeed, Windows is looking for Nu (shimmed) in `/chocolatey/bin` thanks to how `PATH` environment variable is set.

So, I added an extra line in `PATH` just above chocolatey to force Windows to look first in Nu's own install folder (see screenshot below).

And now! It's fixed! No more crashes!!! :D

![image](https://user-images.githubusercontent.com/47865544/187065342-d3102d3a-12dc-4504-981b-71982493b070.png)

**Comment by rgwood** [comment]
Nice, glad you have a solution! Sounds like a bug in Chocolatey; let's leave this issue open for now.

**Comment by Sanceilaks** [comment]
> OK thank you! Your message actually helped me to solve this problem.
> 
> I first tested your simple code and it was crashing too. So it was definitely not related to my Discord bot's code.
> 
> Then I read the documentation you shared about Chocolatey's shim and was _"Hu? What is that?"_ From there, I investigated and found that, indeed, Windows is looking for Nu (shimmed) in `/chocolatey/bin` thanks to how `PATH` environment variable is set.
> 
> So, I added an extra line in `PATH` just above chocolatey to force Windows to look first in Nu's own install folder (see screenshot below).
> 
> And now! It's fixed! No more crashes!!! :D
> 
> ![image](https://user-images.githubusercontent.com/47865544/187065342-d3102d3a-12dc-4504-981b-71982493b070.png)

it's working, lol

**Comment by Afeather2017** [comment]
Yeah, remove nu installed by choco, and install the one in github, then fixxed.
By the way, commands like ncat, ssh(before login), etc, crash too when press CTRL+C.

**Comment by sholderbach** [comment]
This seems to be a choice `chocolatey` made for their shimming mechanism to turn a `Ctrl-C` into a fatal kill:
- https://github.com/chocolatey/home/issues/134

There seems to be workarounds to make chocolatey not use this shimming:
- https://github.com/digitalcoyote/chocolatey-packages/pull/42
- https://github.com/vmware-tanzu/tanzu-cli/pull/665

But as we (the Nushell organisation, or the core team and its members) don't maintain this package, please raise this with the repository that provides the packaging source/recipe for it:
- https://community.chocolatey.org/packages/nushell
- https://github.com/dgalbraith/chocolatey-packages/tree/master/automatic/nushell

**Comment by sholderbach** [comment]
@Afeather2017 for reference, do you remember which version of the choco package you uninstalled?

**Comment by Afeather2017** [comment]
> [@Afeather2017](https://github.com/Afeather2017) for reference, do you remember which version of the choco package you uninstalled?

Sorry for my late repling. Version of choco is 2.4.0. And nushell, I went throught the cache, so I think it is 0.106.1

Hope these could be helpful.

choco:
```
C:\Users\Afeather>choco --version
2.4.0
```
and nushell:
```
 C:\ProgramData\ChocolateyHttpCache\9310b7b8ea69100d3cda3ea6c4468d664f9337fa$community.chocolatey.org_api_v2_ 的目录

2025/08/26  13:54               533 list_Packages()_$filter=(tolower(Id)%20eq%20'nu')%20and%20IsLatestVersion&semVerLevel=2.0.0_page1.dat
2025/06/17  18:02               533 list_Packages()_$filter=(tolower(Id)%20eq%20'nu-shell')%20and%20IsLatestVersion&semVerLevel=2.0.0_page1.dat
2025/08/06  13:49             7,580 list_Packages()_$filter=(tolower(Id)%20eq%20'nushell')%20and%20IsLatestVersion&semVerLevel=2.0.0_page1.dat
2025/08/06  13:49             7,092 list_Packages(Id='nushell',Version='0.106.1')_page1.dat
2025/08/06  13:49             7,116 list_Packages(Id='nushell.install',Version='0.106.1')_page1.dat
```

**Comment by AceCoderLaura** [comment]
I had this problem, installing the winget version fixed it 👍
